deeprank/learn/DataSet.py

```diff
@@ -527,6 +493,29 @@ def _select_pdb(self, mol_names):
 
         return selected_mol_names
 
+    @staticmethod
+    def _insert_before_operators(subject_string, inserting_string):
+        return_string = subject_string
+        for operator_string in ['>=', '<=', '==', '>', '<']:
+            search_index = 0
+            while search_index < len(return_string):
+                found_index = return_string.find(operator_string, search_index)
+
+                if found_index >= 0:
+                    # ATTENTION: we don't want to insert it twice!
+                    if return_string[:found_index].endswith(inserting_string):
+
+                        search_index = found_index + len(operator_string)
+                    else:
+                        return_string = return_string[:found_index] + inserting_string + return_string[found_index:]
+
+                        search_index = found_index + len(inserting_string) + len(operator_string)
+                else:
+                    # not found
+                    break
+
```

Comment:
What is this function useful for?

Reply:
In the original deeprank, one of the authors appears to have made a "filter" function. In this filter function, the variable name "val" appears to be placed before the operators: '>=', '<=', '==', '>', '<'. When I was testing this function however, I bumped into a bug, caused by the operator string not getting detected properly, resulting in an error. That's why I made the function "_insert_before_operators". The purpose of this function is to make sure that the string "val" gets inserted properly before any of the operators. I'm adding a docstring..